tests/test_core.py

```diff
@@ -1,42 +1,96 @@
 import random
 import time
 
-import stbt
+from stbt import wait_until
 
 
-def test_that_wait_until_returns_on_success():
-    count = [0]
+def f(spec):
+    """Helper function for wait_until selftests. Creates a function that returns
+    the specified values one by one (10ms apart) each time it is called.
+
+    Values are specified as space-separated characters. `.` means `None` and
+    `F` means `False`.
 
-    def t():
-        count[0] += 1
-        return True
+    >>> g = f("F a b")
+    >>> [g(), g(), g(), g(), g()]
+    [False, 'a', 'b', False, 'a']
+    """
+    mapping = {".": None, "F": False}
+    values = [mapping.get(x, x) for x in spec.split()]
+    i = [0]
 
-    assert stbt.wait_until(t)
-    assert count == [1]
+    def inner():
+        time.sleep(0.01)
+        if i[0] < len(values):
+            v = values[i[0]]
+            i[0] += 1
+            return v
+        else:
+            i[0] = 1
+            return values[0]
+
+    return inner
+
+
+def test_that_wait_until_returns_on_success():
+    assert wait_until(f(". a b")) == "a"
+    assert wait_until(f("F a b")) == "a"
 
 
 def test_that_wait_until_times_out():
     start = time.time()
-    assert not stbt.wait_until(lambda: False, timeout_secs=0.1)
+    assert wait_until(lambda: False, timeout_secs=0.1) is False
     end = time.time()
     assert 0.1 < end - start < 0.2
 
 
 def test_that_wait_until_tries_one_last_time():
-    def t():
-        time.sleep(0.2)
-        yield False
-        yield True
+    assert wait_until(f("F T"), timeout_secs=0.01) == "T"
+    assert wait_until(f("F F T"), timeout_secs=0.01) is False
+
 
-    assert stbt.wait_until(t().next, timeout_secs=0.1)
+def test_that_wait_until_tries_one_last_time_with_consecutive_secs():
+    #                      consecutive_secs reached
+    #           Timeout reached     |
+    #                         |     |
+    #                         v     v
+    assert wait_until(f("a b b b b b b b b b b"),
+                      timeout_secs=0.03, consecutive_secs=0.05) is None
```

Comment:
This comes as a surprise to me.  I was expecting that the semantics was it must have got stable within the timeout, but they are it must have begun getting stable within the timeout.  Not staying that the current semantics are wrong, just surprising to me.